main.cpp

```diff
@@ -78,21 +78,44 @@ send me a DM to check your pull request
 */
 
 #include <iostream>
-#include <cmath>
-#include <memory>
+#include <math.h>
 #include <functional>
+#include <memory>
 #include <typeinfo>
+#include "LeakedObjectDetector.h"
 
 
 template<typename NumericType>
 struct Temporary
 {
+    // Constructor when passing a NumericType object
     Temporary(NumericType t) : v(t)
     {
         std::cout << "I'm a Temporary<" << typeid(v).name() << "> object, #"
                   << counter++ << std::endl;
     }
-    
+
+    // -------------- The magnificient 5 --------------
+    // Destructor
+    ~Temporary() = default;
+    // Copy constructor (deleted as object non copyable)
+    Temporary(Temporary& other) = delete;
+    // Copy assignment operator (deleted as object non copyable)
+    Temporary& operator=(Temporary& other) = delete;    
+    // Move constructor
+    Temporary(Temporary&& other) : v( std::move(other.v) ) { }
+    // Move assignment operator
+    Temporary& operator=(Temporary&& other)
+    {
+        this->v =  std::move(other.v);
+        return *this;
+    }    
```

Comment:
- [ ] Fixme: Look at what the leak-detector automatically creates for you and revise accordingly.